### PR TITLE
[SW-1301] Fix class-loading for Sparkling Water assembly JAR in PySparkling

### PR DESCRIPTION
**Pull request by jakubhava**
Going to manually test on Databricks Cloud(DBC)

### CODE REVIEW

File: py/pysparkling/initializer.py [hunk 1]
```diff
@@ -46,21 +45,13 @@ def __add_sparkling_jar_to_spark(sc):
...
-        cl = jvm.Thread.currentThread().getContextClassLoader()
 
-        # Explicitly check if we run on databricks cloud since there we must add the jar to
-        # the parent of context class loader
-        if cl.getClass().getName() == 'com.databricks.backend.daemon.driver.DriverLocal$DriverLocalClassLoader':
-            cl.getParent().getParent().addURL(url)
-        else:
-            spark_cl = Initializer.__find_spark_cl(cl, 'org.apache.spark.util.MutableURLClassLoader')
-            spark_cl.addURL(url)
+        loader = Initializer.__find_spark_cl(jvm)
```
Comment (mn-mikke) [inline py/pysparkling/initializer.py]: @jakubhava WDYT about adding the url into each class loader from the hierarchy that has the `addUrl` method. I'm just thinking about cases where Spark is breaking standard lookup from parents to children. E.g. ```ExecutorClassLoader```.

Comment (jakubhava) [reply py/pysparkling/initializer.py]: Good idea! Will do

Comment (mn-mikke) [reply py/pysparkling/initializer.py]: Just one more note to that: I would also add some logging... (the list of loaders where url was added or log an error if there isn't such a classloader)

**Comment by jakubhava** [comment]
@mn-mikke please have a second look, added the logging

File: py/pysparkling/initializer.py [hunk 2]
```diff
@@ -118,11 +108,18 @@ def __get_sw_jar():
...
+    def __add_url_to_classloader(jvm, url):
+        cl = jvm.Thread.currentThread().getContextClassLoader()
         while cl:
-            name = cl.getClass().getName()
-            if (name == cl_name):
-                return cl
+            methods = [m.getName() for m in cl.getClass().getDeclaredMethods()]
+            if "addURL" in methods:
+                cl.addURL(url)
+                Initializer.__get_logger(jvm).info("Adding " + url.toString() + " to classloader"
```
Comment (mn-mikke) [inline py/pysparkling/initializer.py]: nit: Would it be possible to use some string interpolation (E.g. `.format(...)`) and have the message at one line?

Comment (jakubhava) [reply py/pysparkling/initializer.py]: Good idea, done!

**Comment by jakubhava** [comment]
Should be good, please have a final look @mn-mikke

File: py/pysparkling/initializer.py [hunk 3]
```diff
@@ -118,11 +108,18 @@ def __get_sw_jar():
...
+        cl = jvm.Thread.currentThread().getContextClassLoader()
+        logger = Initializer.__get_logger(jvm)
         while cl:
-            name = cl.getClass().getName()
-            if (name == cl_name):
-                return cl
+            methods = [m.getName() for m in cl.getClass().getDeclaredMethods()]
+            if "addURL" in methods:
+                cl.addURL(url)
+                logger.info("Adding {} to classloader '{}'".format(url.toString(), cl.toString()))
```
Comment (mn-mikke) [inline py/pysparkling/initializer.py]: This  looks great. I'm just wondering whether we shouldn't use the debug logging level instead. WDYT?

Comment (jakubhava) [reply py/pysparkling/initializer.py]: yup

**Pull request merged**